src/Plotly.NET/ChartAPI/Chart2D.fs

```diff
@@ -6014,4 +6014,530 @@ module Chart2D =
+                ?TextFont: Font,
+                ?UseDefaults: bool
+            ) =
+
+            let useDefaults =
+                defaultArg UseDefaults true
+
+            let referenceName = defaultArg ReferenceName "Reference Line"
+
+            // if text position or font is set than show labels (not only when hovering)
+            let changeMode =
+                let isShowMarker =
+                    match ShowMarkers with
+                    | Some isShow -> isShow
+                    | Option.None -> false
+
+                StyleParam.ModeUtils.showText (TextPosition.IsSome || TextFont.IsSome)
+                >> StyleParam.ModeUtils.showMarker (isShowMarker)
+
+            let showReference = 
+                let showBool = 
+                    match ShowReference with
+                    | Some showRef   -> showRef
+                    | None                  -> true
+                if showBool then 
+                    StyleParam.Visible.True
+                else
+                    StyleParam.Visible.False
+
+            let markerSize = defaultArg MarkerSize 10
+            
+            let (data,pointS) = 
+                Seq.zip3 x y reference
+                |> Seq.collect(fun (x,y,refV) ->
+                    [
+                        (x,refV),0
+                        (x,y),markerSize
+                        (x,refV),0
+                    ]
+                )
+                |> Array.ofSeq
+                |> Array.unzip
+ 
+            let (x1,y1) = Array.unzip data
+            
+            let marker = 
+                defaultArg 
+                    Marker 
+                    (TraceObjects.Marker.init(
+                        MultiSize=pointS,
+                        Opacity=1.,
+                        ?Color = MarkerColor,
+                        ?Colorscale = MarkerColorScale,
+                        ?Outline = MarkerOutline,
+                        ?Symbol = MarkerSymbol,
+                        ?MultiSymbol = MultiMarkerSymbol
+                    ))
+
+            let trace =
+                Chart.Scatter(
+                    x = x1,
+                    y = y1,
+                    mode = changeMode StyleParam.Mode.Lines_Markers,
+                    Opacity = 1.,
+                    Marker = marker,
+                    ?Name = Name,
+                    ?ShowLegend = ShowLegend,
+                    ?Text = Text,
+                    ?MultiText = MultiText,
+                    ?TextPosition = TextPosition,
+                    ?MultiTextPosition = MultiTextPosition,
+                    ?LineColor = LineColor,
+                    ?LineColorScale = LineColorScale,
+                    ?LineWidth = LineWidth,
+                    ?LineDash = LineDash,
+                    ?Line = Line
+                )
+                |> GenericChart.mapTrace (
+                    Trace2DStyle.Scatter(
+                        LegendGroup = (defaultArg GroupName "Datapoints"),
```

Comment:
To clarify from comment above, just pass `?LegendGroup = GroupName`, so the userchoice is respected, but no default is set. Same for any other legend group settings.